core/domain/auth_validators.py

```diff
@@ -0,0 +1,58 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for auth models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import base_model_validators
+from core.platform import models
+
+(auth_models, user_models) = models.Registry.import_models(
+    [models.NAMES.auth, models.NAMES.user])
+
+
+class UserIdByFirebaseAuthIdModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating UserIdByFirebaseAuthIdModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        """Returns a regex for model id.
+
+        This method can be overridden by subclasses, if needed.
+
+        Args:
+            unused_item: datastore_services.Model. Entity to validate.
+
+        Returns:
+            str. A regex pattern to be followed by the model id.
+        """
+        return '^[\x00-\x7F]{1,255}$'
```

Comment:
Perhaps add a source/rationale for this pattern. E.g. can it really be of length 1? It should be as strict as possible.

Reply:
Done, and yep it really can be of length 1: https://firebase.google.com/docs/auth/admin/manage-users#create_a_user.